packages/babel-plugin-proposal-class-properties/test/fixtures/general/derived/expected.js

```diff
@@ -7,7 +7,7 @@ function (_Bar) {
     var _temp, _this;
 
     babelHelpers.classCallCheck(this, Foo);
-    return babelHelpers.possibleConstructorReturn(_this, (_temp = _this = babelHelpers.possibleConstructorReturn(this, (Foo.__proto__ || Object.getPrototypeOf(Foo)).call(this, ...args)), Object.defineProperty(babelHelpers.assertThisInitialized(_this), "bar", {
+    return babelHelpers.possibleConstructorReturn(_this, (_temp = _this = babelHelpers.possibleConstructorReturn(this, babelHelpers.constructSuperInstance(Foo, [...args], this)), Object.defineProperty(babelHelpers.assertThisInitialized(_this), "bar", {
```

Comment:
Possible future microoptimization: `[...args]` -> `args`

Only possible if we can prove `args` is arraylike, though 🤷‍♀️

Reply:
Is there something that prevents doing it now? We could use `Array.isArray` and its helper.

Reply:
@xtuc the idea is to do the optimization statically, though. It'd also miss out on arraylike that are not arrays